### ISSUE TITLE
Add feature to "mock once" an implementation.

### ISSUE THREAD

**Issue by sirbeefalot** [EthWorks/Waffle#392]
There are situations where a contract function will call the same mock function twice, and it's required for both calls to give two different results for the execution to work.

Is there any way to chain a couple mockOnce() before starting a test transaction?

Thanks

**Comment by gbarros** [comment]
Yes, this is a very common use case. For instance, you want to check on a contract the balance change after a call, so likely you will call `balanceOf` twice.

**Comment by steffenix** [comment]
+1
